Framework/Indexing/src/IndexInfo.cpp

```diff
@@ -226,47 +226,59 @@ SpectrumIndexSet IndexInfo::makeIndexSet(
   return m_spectrumNumberTranslator->makeIndexSet(globalIndices);
 }
 
-/** Map a vector of detector indices to a vector of global spectrum indices.
- *
- * The mapping is based on the held spectrum definitions. Throws if any spectrum
- * maps to more than one detectors. Throws if there is no 1:1 mapping from
- * detectors to spectra, such as when some of the detectors have no matching
- * spectrum. */
+/** Map a vector of detector indices to a vector of global spectrum indices. *
+ * The mapping is based on the held spectrum definitions.
+ * Throws if any spectrum maps to more than one detectors.
+ * Throws if any of the detectors has no matching spectrum.
+ * Throws if more than one spectrum maps to the same detector at the same time
+ * index.
+ */
 std::vector<GlobalSpectrumIndex>
 IndexInfo::globalSpectrumIndicesFromDetectorIndices(
     const std::vector<size_t> &detectorIndices) const {
   if (!m_spectrumDefinitions)
     throw std::runtime_error("IndexInfo::"
                              "globalSpectrumIndicesFromDetectorIndices -- no "
                              "spectrum definitions available");
-  std::vector<char> detectorMap;
+  /*
+   * We need some way of keeping track of which time indices of given detector
+   * have a matching mapping. detectorMap holds pairs; first in the pair
+   * indicates the detector ID, used to check if there is mapping with the
+   * detector, regardless of time index. The second, which is a vector indexed
```

Comment:
Can you also add a comment further down, explaining that initially `first` is 1 if the detector was requested, but is then incremented if matches were found.